### ISSUE TITLE
Child closed pipe

### ISSUE THREAD

**Issue by CreasaBE** [ddserver/ddserver#84]
Hi,

I'm trying to set this up, but when starting powerdns, I get the following error:

```
Apr 11 20:04:26 vipbackup01 pdns[476]: [PIPEBackend] Fatal argument error: Child closed pipe
Apr 11 20:04:26 vipbackup01 pdns[476]: Caught an exception instantiating a backend: Child closed pipe
Apr 11 20:04:26 vipbackup01 pdns[476]: Cleaning up
Apr 11 20:04:26 vipbackup01 pdns[476]: Exiting because communicator thread died with error: Child closed pipe
Apr 11 20:04:26 vipbackup01 pdns[476]: [PIPEBackend] Fatal argument error: Child closed pipe
Apr 11 20:04:26 vipbackup01 pdns[476]: Caught an exception instantiating a backend: Child closed pipe
Apr 11 20:04:26 vipbackup01 pdns[476]: Cleaning up

```

Did I forget something, or is something broken with the latest powerdns (running Debian Jessie)

exerpt from my pdns.conf file

```
launch=gmysql,pipe
pipe-command=/usr/local/bin/ddserver-recursor

```

```
# ls -l /usr/local/bin/ddserver-*
-rwxr-xr-x 1 root root 314 Apr 11 19:21 /usr/local/bin/ddserver-bundle
-rwxr-xr-x 1 root root 320 Apr 11 19:21 /usr/local/bin/ddserver-interface
-rwxr-xr-x 1 root root 322 Apr 11 19:47 /usr/local/bin/ddserver-recursor
-rwxr-xr-x 1 root root 316 Apr 11 19:21 /usr/local/bin/ddserver-updater

```

Thanks.

**Comment by fooker** [comment]
Can you try to run the recursor manually by executing the command as given in the config (`/usr/local/bin/ddserver-recursor` in your example)? If it starts successfully, you can try to type `PING` followed by a newline.

**Comment by CreasaBE** [comment]
the recursor starts when started manually

```
 # /usr/local/bin/ddserver-recursor
PING
recursor: Received message: PING(tag='PING')
recursor: Missing HELO before command: PING
recursor: Responding message: FAIL(tag='FAIL')
FAIL
```

The log contains the following

```
recursor: Received message: PING(tag='PING')
recursor: Missing HELO before command: PING
recursor: Responding message: FAIL(tag='FAIL')
recursor: Received message: PING(tag='PING')
recursor: Missing HELO before command: PING
recursor: Responding message: FAIL(tag='FAIL')
```

**Comment by CreasaBE** [comment]
Hi,

Any news? Is there something more I can try or would you happen to know where to search for more debugging?

Thanks

**Comment by reissmann** [comment]
There could be an issue with the PYTHON_PATH variable. Could you please try to start powerdns in debug mode, by first stopping the daemon and then run this command:

```
pdns_server
```

**Comment by CreasaBE** [comment]
got it.
after enabling extensive debugging and disabling daemon mode, I found that it failed on the python_egg cache directory. It wanted to use the root's directory, but off course it could not.
After adding "export PYTHON_EGG_CACHE=<path to the pdns home folder>/python-eggs/" to the defaults file (/etc/default/pdns) it works.

Tested by adding a domain, hostname and the hostname resolves. 
Seems all is ok now.

Thanks for the help. (maybe this is something that can be added to the manual/troubleshooting)

grtz
